sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/resource/Ec2Resource.java

```diff
@@ -206,4 +206,16 @@ private String fetchIdentity(String token) {
   private String fetchHostname(String token) {
     return fetchString("GET", hostnameUrl, token, /* includeTtl= */ false);
   }
+
+  private static void readTo(@Nullable InputStream is, ByteArrayOutputStream os)
+      throws IOException {
+    if (is == null) {
+      return;
+    }
+    byte[] buf = new byte[8192];
+    int read;
+    while ((read = is.read(buf)) != -1) {
```

Comment:
```suggestion
    while ((read = is.read(buf)) > 0) {
```

Although -1 should also work according to <https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A->, I think this is slightly more clear.